pkg/serverstate/statetest/test_job.go

```diff
@@ -2681,9 +2681,11 @@ func TestJobHeartbeatOnRestart(t *testing.T, factory Factory, rf RestartFactory)
 		defer s.Close()
 
 		// Verify it exists
```

Comment:
```suggestion
		// Verify it exists. We use a closure here to avoid a possible race condition with the job object
```

nitpick: it's not immediately obvious that we use a closure here so we can make a new `job` object to avoid the race condition, so just add a comment incase someone finds it odd, or happens to copy/paste this test or parts of it for use in future tests